modules/nextflow/src/main/groovy/nextflow/script/FunctionNameDef.groovy

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020-2022, Seqera Labs
+ * Copyright 2013-2019, Centre for Genomic Regulation (CRG)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nextflow.script
+
+import groovy.transform.CompileStatic
+
+import javax.mail.MethodNotSupportedException
+import java.lang.reflect.Method
+
+import static ChannelOut.spreadToArray
+
+/**
+ * Models a function component that can be included from a module script
+ *
+ * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
+ */
+
+@CompileStatic
+class FunctionNameDef extends ComponentDef implements ChainableDef {
```

Comment:
I think it would be better to modify `FunctionDef` instead of adding a new class. it's meant exactly for this kind of stuff